docs/reference/data-streams/logs.asciidoc

```diff
@@ -46,154 +50,143 @@ DELETE _index_template/my-index-template
 [[logsdb-default-settings]]
 
 [discrete]
-[[logsdb-synthtic-source]]
+[[logsdb-synthetic-source]]
 === Synthetic source
 
-By default, `logsdb` mode uses <<synthetic-source,synthetic source>>, which omits storing the original `_source`
-field and synthesizes it from doc values or stored fields upon document retrieval. Synthetic source comes with a few
-restrictions which you can read more about in the <<synthetic-source,documentation>> section dedicated to it.
+If you have the required https://www.elastic.co/subscriptions[subscription], `logsdb` index mode uses <<synthetic-source,synthetic `_source`>>, which omits storing the original `_source`
+field. Instead, the document source is synthesized from doc values or stored fields upon document retrieval. 
+
+Before using synthetic source, make sure to review the <<synthetic-source,restrictions>>. 
 
-NOTE: When dealing with multi-value fields, the `index.mapping.synthetic_source_keep` setting controls how field values
+When working with multi-value fields, the `index.mapping.synthetic_source_keep` setting controls how field values
 are preserved for <<synthetic-source,synthetic source>> reconstruction. In `logsdb`, the default value is `arrays`,
-which retains both duplicate values and the order of entries but not necessarily the exact structure when it comes to
-array elements or objects. Preserving duplicates and ordering could be critical for some log fields. This could be the
-case, for instance, for DNS A records, HTTP headers, or log entries that represent sequential or repeated events.
+which retains both duplicate values and the order of entries. However, the exact structure of
+array elements and objects is not necessarily retained. Preserving duplicates and ordering can be critical for some log fields, such as DNS A records, HTTP headers, and log entries that represent sequential or repeated events. 
 
-For more details on this setting and ways to refine or bypass it, check out <<synthetic-source-keep, this section>>.
+If you don't have the required https://www.elastic.co/subscriptions[subscription], `logsdb` mode uses the original `_source` field.
 
 [discrete]
 [[logsdb-sort-settings]]
 === Index sort settings
 
-The following settings are applied by default when using the `logsdb` mode for index sorting:
+In `logsdb` index mode, the following sort settings are applied by default:
 
-* `index.sort.field`: `["host.name", "@timestamp"]`
-  In `logsdb` mode, indices are sorted by `host.name` and `@timestamp` fields by default. For data streams, the
-  `@timestamp` field is automatically injected if it is not present.
+`index.sort.field`: `["host.name", "@timestamp"]`::
```

Comment:
I was wondering if there is a good way to highlight such settings around index sorting applied by default in LogsDB and the **potential that they might cause integration issues when upgrading**, especially for custom integrations. We know that our (Elastic) integrations follow ECS and we do not have conflicts when it comes to mapping the `host.name` field but I think it is quite important to highlight that having this field automatically mapped might potentially cause issues for custom integrations (think of a customer adopting LogsDB for the first time with a new release of Elasticsearch and having a `host.name` field being a `text` field for instance).

At the moment what happens is the following: if index sort settings include `host.name` a `host.name` field will be automatically mapped as a keyword (see below for details on what is actually mapped depending on the value of `subobjects`). Because of this behaviour a customer adopting LogsDB for the first time while upgrading to the new version might experience mapping issues in case they have existing mappings using a `host.name` field mapped with a different type. This conflict leads to mapping errors, and mapping errors might result in not being able to create new indices.

IMO opinion we need to clearly state for customers transitioning to using LogsDB and adopting it for the first time, that they have at least three options here:
1. make sure, if they already have a `host.name` field, that it is mapped as a `keyword` before upgrading
2. remove `host.name` field from the set of sort fields so to avoid  `host.name` field being automatically injected by Elasticsearch (we inject the `host.name` mapping only if it is part of fields to sort on. Ideally they should choose a suitable set of fields to sort on, excluding `host.name` and maybe go for only sorting by `@timestamp` as a last-resort option).
3. opt-out of LogsDB in case there is no other way to deal with a potential mapping conflict.

I just want to restate this is IMO **critical** to highlight in the documentation since neglecting this detail might result in upgrade issues for customers moving from standard indices to LogsDB as a result of upgrading. Is there a good way to highlight this so to reduce the chances for customers to hit such a problem?

Reply:
Thanks, will work on adding some of this